config/user-libudev.m4

```diff
@@ -0,0 +1,13 @@
+dnl #
+dnl # Check for libudev - needed for vdev auto-online and auto-replace
+dnl #
+AC_DEFUN([ZFS_AC_CONFIG_USER_LIBUDEV], [
+	LIBUDEV=
+
+	AC_CHECK_HEADER([libudev.h], [AC_SUBST([LIBUDEV], ["-ludev"])
+	AC_DEFINE([HAVE_LIBUDEV], 1, [Define if you have libudev])], [])
+
+	AC_CHECK_LIB([udev], [udev_device_get_is_initialized], [
```

Comment:
Understood.  I think the value is in having .m4 files be good examples (admittedly I borrowed from another one). In theory I can just drop in `AC_SEARCH_LIBS` macro and swap the first two arguments.